dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/DotParse.java

```diff
@@ -1,159 +1,239 @@
-
-      //We found the resource but not the version we are looking for
-      if ( null == inode ) {
-          throwNotResourceNotFoundException(params, templatePath);
-      }
-
-      final boolean respectFrontEndRolesForVTL = params.mode.respectAnonPerms?
-              Config.getBooleanProperty("RESPECT_FRONTEND_ROLES_FOR_DOTPARSE", true) : params.mode.respectAnonPerms;
-      final Contentlet contentlet = APILocator.getContentletAPI().find(inode, APILocator.getUserAPI().getSystemUser(), respectFrontEndRolesForVTL);
-      final FileAsset asset = APILocator.getFileAssetAPI().fromContentlet(contentlet);
-
-      // add the edit control if we have run through a page render
-      if (!context.containsKey("dontShowIcon") &&
-              PageMode.EDIT_MODE == params.mode ) {
-          final String editIcon = String.format(EDIT_ICON, contentlet.getInode(), identifier.getURI(),
-                  APILocator.getPermissionAPI().doesUserHavePermission(contentlet, PermissionAPI.PERMISSION_READ, user),
-                  APILocator.getPermissionAPI().doesUserHavePermission(contentlet, PermissionAPI.PERMISSION_EDIT, user));
-          writer.append(editIcon);
-      }
-
-      return null != asset.getFileAsset()? asset.getFileAsset().getAbsolutePath():null;
-    } catch (ResourceNotFoundException e) {
-        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-        if(e.getStackTrace().length>0){
-          Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
+
+        final StringTokenizer tokens = new StringTokenizer(templatePath, StringPool.FORWARD_SLASH);
+
+        if (tokens.countTokens() < 2) {
+            String errorMessage = String.format("No resource found for [%s]", templatePath);
+            throw new ResourceNotFoundException(errorMessage);
         }
-        throw e;
-    } catch (DotSecurityException  e) {
-        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-        if(e.getStackTrace().length>0){
-            Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
+
+        tokens.nextToken();
+        final String inodeOrIdentifier = tokens.nextToken();
+        final String fieldVar = tokens.hasMoreTokens() ? tokens.nextToken() : DotAssetContentType.ASSET_FIELD_VAR;
+        final Optional<ShortyId> shortOpt = APILocator.getShortyAPI().getShorty(inodeOrIdentifier);
+
+        if (!shortOpt.isPresent()) {
+            return Optional.empty();
         }
-        throw new ResourceNotFoundException(e);
-    } catch (Exception e) {
-        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-        if(e.getStackTrace().length>0){
-            Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
+
+        ShortyId shorty = shortOpt.get();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'shorty' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5627236)